lib/src/dlis/io.cpp

```diff
@@ -514,6 +514,29 @@ dl::error_handler& errorhandler) noexcept (false) {
     return xs;
 }
 
+std::vector< dl::record >
```

Comment:
I'm not convinced about this move. Here it is a part of the C++ API, and I don't think we want that. The purpose of this function is not _convenience_, it's _performance_. It avoids going back and forth between python and C++. I don't think we should expose this function for convenience, as people can write their own loops for their specific purposes.